paddle/fluid/distributed/collective/ProcessGroupMPI.cc

```diff
@@ -0,0 +1,426 @@
+// Copyright (c) 2022 PaddlePaddle Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "paddle/fluid/distributed/collective/ProcessGroupMPI.h"
+#include <chrono>
+#include "paddle/fluid/distributed/collective/Common.h"
+#include "paddle/fluid/platform/cuda_device_guard.h"
+
+constexpr int64_t kWaitBlockTImeout = 10;
+std::map<phi::DataType, MPI_Datatype> mpiDatatype = {
```

Comment:
Global var?

Reply:
if regard it as a member of ProcessGroupMPI class, it  may be static member,  otherwise， it will be a duplicate in every  ProcessGroupMPI object. as a global map tool for mpi data type in distributed namespace， when backend changed, it may be come in handy.